### PR TITLE
Option to unpublish without stopping track

### PR DESCRIPTION
**Pull request by hiroshihorie**
Now there is a global option and also overridable at `unpublishTrack` and `unpublishAllTracks`

https://linear.app/livekit/issue/LK-308/api-design-unpublishleave-room-without-stopping-local-tracks

Does it make sense for Local tracks only ?

### CODE REVIEW

**Comment by hiroshihorie** [comment]
Would appreciate your thoughts about this design @davidzhao @davidliu 🙂

**Comment by davidliu** [comment]
Seems like it'd only be for local tracks right? I can't think of cases where the track is unpublished, but the track can still technically be subscribed to and receive any track data.

**Comment by hiroshihorie** [comment]
@davidliu Yes I think it makes sense only for Local tracks 😂

File: lib/src/participant/local_participant.dart [hunk 1]
```diff
@@ -170,7 +170,11 @@ class LocalParticipant extends Participant {
 
   /// Unpublish a track that's already published
   @override
-  Future<void> unpublishTrack(String trackSid, {bool notify = true}) async {
+  Future<void> unpublishTrack(
+    String trackSid, {
+    bool notify = true,
+    bool? stopTrack,
```
Comment (davidzhao) [inline lib/src/participant/local_participant.dart:176]: I wonder if we should offer both defaults *and* a parameter for each unpublish. what if we didn't have a room option; would it change how users would use this?

File: lib/src/participant/remote_participant.dart [hunk 2]
```diff
@@ -175,7 +175,11 @@ class RemoteParticipant extends Participant {
   }
 
   @override
-  Future<void> unpublishTrack(String trackSid, {bool notify = true}) async {
+  Future<void> unpublishTrack(
+    String trackSid, {
+    bool notify = true,
+    bool? stopTrack,
```
Comment (davidzhao) [inline lib/src/participant/remote_participant.dart:181]: as DL mentioned, I don't think this makes sense for remote. ideally this is exposed only to the package and not outside of the library.

**Comment by hiroshihorie** [comment]
I have reverted changes, now no option on `unpublishTrack`
engine.connectOptions is never null so there will always be default option.

**Review by davidzhao — APPROVED** [review]
looks good!

**Pull request merged**